### DISCUSSION TITLE
A replication sw

### DISCUSSION THREAD

**Discussion by tcapturesupport** [lab-sb/tcrepsrv-dev#5] [General]
Born from the need to have a simple and effective tool for data replication between different Postgres versions.
This is achieved by logical decoding, i.e. the ability to direct sql changes to consumers.
The plugin used for this purpose captures the transactions applied to the source, directs them into queues specifically built to be consumed by one or more consumers.

**Comment by tcapturesupport** [comment]
![image](https://github.com/lab-sb/tcrepsrv-dev/assets/146429786/1d975d49-71b8-4f8f-af3d-b48989182cae)

**Comment by Dave-Cooke** [comment]
Greetings.  Our software uses postgres10 and I have been able to use the built-in tools to replicate a server backup, but I am wondering if TCapture would be able to have bidirectional replication?  I am okay with last-out winning the conflict, but with my current setup SQL server A overwrites SQL server B with an exact copy, but if I make a change on B it is overwritten by A instead of syncing.  Thanks for your time to answer my question.

**Comment by lab-sb** [comment]
Hi Dave,

you can configure TCapture in bidirectional replication  ,  something like the following:


 -- sconfigure
sh TC_srvctl.sh  --config --type producer --node NODE-A  --host PSQLSERVER-A --port 5432 --user repusr --passwd reppwd -db databasename-A --rhost  PSQLSERVER-A  --ruser repusr --rport 5432 --rpasswd reppwd
sh TC_srvctl.sh  --config --type consumer --node NODE-B  --host PSQLSERVER-B   --port 5432  --user repusr --passwd reppwd -db databasename-A  --rhost PSQLSERVER-B --ruser repusr --rport 5432 --rpasswd reppwd
 
 sh TC_srvctl.sh  --config --type producer --node NODE-B  --host PSQLSERVER-B --port 5432 --user repusr --passwd reppwd -db databasename-B --rhost  PSQLSERVER-B  --ruser repusr --rport 5432 --rpasswd reppwd
sh TC_srvctl.sh  --config --type consumer --node NODE-A  --host PSQLSERVER-A   --port 5432  --user repusr --passwd reppwd -db databasename-A  --rhost PSQLSERVER-A --ruser repusr --rport 5432 --rpasswd reppwd
 
 -- setup 
 sh TC_srvctl.sh --setup  --node NODE-A --type producer
sh TC_srvctl.sh --setup  --node NODE-B --type consumer --producer NODE-A 
 
 
 sh TC_srvctl.sh --setup  --node NODE-B --type producer
sh TC_srvctl.sh --setup  --node NODE-A --type consumer --producer NODE-B

-- run the 2 master ( also configured as slave of the other master)

 sh bin/runTCRepSrv.sh -n NODE-B
  sh bin/runTCRepSrv.sh -n NODE-A

**Comment by Dave-Cooke** [comment]
Thank you for your swift reply.  I have never used TCapture and that looks
a little daunting, but manageable.  I do want to ask a clarifying
question.

Just looking at the pattern of A's and B's, should the database name be B
instead of A in the Node B config?  It was all A's in the first config.

sh TC_srvctl.sh --config --type consumer --node NODE-B --host PSQLSERVER-B
--port 5432 --user repusr --passwd reppwd -db *databasename-A* --rhost
PSQLSERVER-B --ruser repusr --rport 5432 --rpasswd reppwd

**Comment by Dave-Cooke** [comment]
Apologies for wasting your time.  I just read the requirements and since
our software is Windows based we won't be able to use TCapture.
But I do again thank you for your reply.

Dave
